### PR TITLE
Extract common webpack config and add browser field to resolve dependencies

### PR DESCRIPTION
**Pull request by victorg1991**
I've left this [file](https://github.com/liferay/clay/blob/master/packages/clayui.com/webpack.config.js) without changing because I'm not sure if it is being used, can you confirm it?

Let me know if something else needs to be changed :)

### CODE REVIEW

**Comment by carloslancha** [comment]
What about making a module with this kind of config files instead having them living in the root folder? @julien @victorg1991 ?

**Comment by julien** [comment]
@carloslancha sounds like a good option to me 👍

**Comment by victorg1991** [comment]
Hey, I've done the changes :) @carloslancha  @julien 

I've also changed the base branch to develop, is that the correct branch, right?

Thanks!

**Comment by carloslancha** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: packages/generator-metal-clay/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "generator-metal-clay",
   "version": "2.1.12",
   "description": "Yeoman generator for Clay components",
-  "license": "MIT",
+  "license": "BSD",
```
Comment (jbalsas) [inline packages/generator-metal-clay/package.json]: Why the license change?

Comment (victorg1991) [reply packages/generator-metal-clay/package.json]: To be in line with other licensed, we thought it was wrong. Should I revert this?

File: packages/browserslist-config-clay-components/package.json [hunk 2]
```diff
@@ -9,5 +9,8 @@
     "clay",
     "browserslist",
     "browserslist-config"
-  ]
+  ],
+  "dependencies": {
+    "clay-config": "^2.1.12"
```
[the remark this answers is not in the document]
Comment (carloslancha) [reply packages/browserslist-config-clay-components/package.json]: I'm already changing this to devDependency in all packages

**Comment by carloslancha** [comment]
Resent https://github.com/liferay/clay/pull/1050

**Pull request closed without merging**